clang/test/SemaSYCL/unnamed-kernel.cpp

```diff
@@ -89,20 +101,31 @@ struct MyWrapper {
 
     using InvalidAlias1 = InvalidKernelName5;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-    // expected-error@Inputs/sycl.hpp:220 {{kernel needs to have a globally-visible name}}
-    // expected-note@24 {{InvalidKernelName5 declared here}}
+    // expected-error@Inputs/sycl.hpp:220 {{invalid type 'MyWrapper::InvalidKernelName5' used in kernel name. Kernel name should be globally-visible}}
+    // expected-note@Inputs/sycl.hpp:220 {{Invalid kernel name is 'namespace1::KernelName<MyWrapper::InvalidKernelName5>'}}
+    // expected-note@30 {{InvalidKernelName5 declared here}}
     // expected-note@+3{{in instantiation of function template specialization}}
 #endif
     q.submit([&](cl::sycl::handler &h) {
       h.single_task<namespace1::KernelName<InvalidAlias1>>([] {});
     });
+#ifndef __SYCL_UNNAMED_LAMBDA__
+    // expected-error@Inputs/sycl.hpp:220 {{invalid type 'InvalidKernelName1' used in kernel name. Kernel name should be globally-visible}}
+    // expected-note@Inputs/sycl.hpp:220 {{Invalid kernel name is 'Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>'}}
+    // expected-note@41 {{InvalidKernelName1 declared here}}
+    // expected-note@+3{{in instantiation of function template specialization}}
+#endif
+    q.submit([&](cl::sycl::handler &h) {
+      h.single_task<Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>>([] {});
+    });
   }
 };
 
 int main() {
   cl::sycl::queue q;
 #ifndef __SYCL_UNNAMED_LAMBDA__
   // expected-error@Inputs/sycl.hpp:220 {{kernel name is missing}}
+  // expected-note-re@Inputs/sycl.hpp:220 {{Invalid kernel name is '(lambda at {{.*}}unnamed-kernel.cpp{{.*}}'}}
```

Comment:
I am not familiar with this usage {{.*}}. Could you please elaborate what it expands to?

Reply:
It is regular expression for any count of any symbol. So, if it is not really important what text is emitted in some parts of the diagnostic (for example here (if I remember correct) it is full path to test directory and source location of the lambda) you can use this thing.

Reply:
One more point - when regular expressions are used, 're' is appended to the 'expected-xyz'